components/movementsensor/cameramono/cameramono_test.go

```diff
@@ -0,0 +1,219 @@
+package cameramono
+
+import (
+	"context"
+	"image"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/edaniels/gostream"
+	"github.com/golang/geo/r3"
+	geo "github.com/kellydunn/golang-geo"
+	"go.uber.org/zap"
+	"go.viam.com/test"
+	"go.viam.com/utils/artifact"
+	"gonum.org/v1/gonum/mat"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/rimage"
+	"go.viam.com/rdk/rimage/transform"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/testutils/inject"
+	"go.viam.com/rdk/vision/keypoints"
+	"go.viam.com/rdk/vision/odometry"
+)
+
+var tCo = &cameramono{
+	Unimplemented:           generic.Unimplemented{},
+	activeBackgroundWorkers: sync.WaitGroup{},
+	cancelCtx:               context.Background(),
+	cancelFunc: func() {
+	},
+	logger: &zap.SugaredLogger{},
+	result: result{
+		trackedPos:    r3.Vector{X: 4, Y: 5, Z: 6},
+		trackedOrient: &spatialmath.OrientationVector{Theta: 90, OX: 1, OY: 2, OZ: 3},
+		angVel:        spatialmath.AngularVelocity{X: 10, Y: 20, Z: 30},
+		linVel:        r3.Vector{X: 40, Y: 50, Z: 60},
+	},
+}
+
+func TestInit(t *testing.T) {
+	camName := "cam"
+	conf := &AttrConfig{}
+	err := conf.Validate("")
+	test.That(t, err.Error(), test.ShouldContainSubstring, "single camera")
+	conf.Camera = camName
+	err = conf.Validate("")
+	test.That(t, err.Error(), test.ShouldContainSubstring, "motion_estimation_config")
+
+	conf.MotionConfig = &odometry.MotionEstimationConfig{
+		KeyPointCfg: &keypoints.ORBConfig{
+			Layers:          1,
+			DownscaleFactor: 2,
+			FastConf: &keypoints.FASTConfig{
+				NMatchesCircle: 1,
+				NMSWinSize:     1,
+				Threshold:      1,
+				Oriented:       true,
+				Radius:         1,
+			},
+			BRIEFConf: &keypoints.BRIEFConfig{
+				N:              1,
+				Sampling:       1,
+				UseOrientation: true,
+				PatchSize:      1,
```

Comment:
If you want better defaults (it might not matter for the tests you're writing)
N: 128
Sampling: 0
PatchSize: 16

Reply:
Thanks! It doesn't matter right now since this is just testing that it made a newCameraMono successfully. But it's a good idea to set this config up for future improvements once we get there.